### DISCUSSION TITLE
Rendering the latest item from a collection on /index.html

### DISCUSSION THREAD

**Discussion by brendon** [11ty/eleventy#2950] [Q&A]
I'm making a very simple news article system where each article is in `/articles/article-*.md`. Articles are rendered with a template `article.html` and the template loops some front matter in each article (a selection of images) and outputs them before the main markdown content is inserted.

This all works well and is straightforward. Unfortunately however I've really hit up against a wall when trying to make `/index.html` render the latest article (as in the last article in the collection).

I can grab the latest article via `collections.article | last` and output `article.templateContent` but this only outputs the markdown and doesn't take into account the template. I've tried adding `layout: article.html` as front matter on `/index.html` but to no avail. It tries to render with the template but doesn't find the variables it needs (e.g. the `images` array that would have come from the front matter in the article.

I'm hoping I've missed something obvious. I've tried using `eleventyComputed` to pre-compute the latest article but no dice.

**Comment by brendon** [comment]
I managed to get it to work but it's gross. Perhaps there's a cleaner way to delegate the front matter?

`index.html`
```
{% assign article = collections.article | last %}
{% assign images = article.data.images %}
{% assign date = article.data.date %}
{% assign content = article.content %}

{% include "_includes/article.html" %}
```

My clue came from here: https://github.com/11ty/eleventy/issues/576#issuecomment-506353172

**Comment by btrem** [comment]
Why use the `assign` blocks to assign individual variables? Instead of

```
{% assign date = article.data.date %}
Posted on {{ date }}
```
I'd just use this:
`Posted on {{ article.date }}`

(N.B.: I use `article.date`, not `article.data.date`. You might have some reason to opt for the latter that I'm unaware of.)

Similarly, I'd just use something like this for content:

```
<main>
{{ article.content }}
</main>
```

If I've misunderstood the problem, let me know.

**Reply by brendon** [reply]
Hi @btrem, I wanted to reuse the `article` template rather than repeat it in `index.html` (thus the `include` call). In `article.html` I was referring directly to the individual attributes of `article` such as `images` and `date`, and that's why I had to `assign` those separately in `index.html`. Certainly if I was duplicating the view code, I could have optimised it as you say.

Essentially I wanted to show the latest article on the homepage of the site, exactly as it would look if one visited it at `/articles/`. There doesn't seem to have been consideration made for this in eleventy so far.

Unfortunately I didn't proceed with my eleventy app as I found a more efficient way to surface these simple articles using a `yaml` file and simple views in my Rails application instead. I did find eleventy fun to use though. I especially liked how it was easy to start from scratch rather than from a massive scaffold that usually doesn't fit (e.g. Jekyll).
